department-of-veterans-affairs/caseflow
# Add validation to hearing table update

Add validation to HEARSHED table update:

1. notes get cut off at length of 100
2. disposition allowed values: :held, :cancelled, :postponed, :no_show, nil
3. aod allowed values: :granted, :filed, :none, nil
4. hold_open must be an integer between 0 and 90 or nil
5. transcript_requested allowed values: false, true, nil

Steps to validate:
Steps to Validate:

1. Start `rails c` and ensure you are connected to Vacols dev env
2. Find a hearing in the hearings table (make sure it exists in Vacols)
```h = Hearing.find(id)```
3. Update any field on the hearing object which will trigger an update on the HEARSHED table. For example,
```
h.update(disposition: :held)
h.update(hold_open: 30)
h.update(aod: :granted)
h.update(notes: "test")
h.update(transcript_requested: false)
```
4. Ensure that validation is working
